src/jit/lowerxarch.cpp

```diff
@@ -2191,8 +2191,26 @@ void Lowering::TreeNodeInfoInitModDiv(GenTree* tree)
         info->setDstCandidates(l, RBM_RAX);
     }
 
-    // If possible would like to have op1 in RAX to avoid a register move
-    op1->gtLsraInfo.setSrcCandidates(l, RBM_RAX);
+#ifdef _TARGET_X86_
+    if (op1->OperGet() == GT_LONG)
+    {
+        GenTree* loVal = op1->gtGetOp1();
+        GenTree* hiVal = op1->gtGetOp2();
+        
+        // Src count is actually 3, so increment.
+        info->srcCount++;
+
+        // Op1's low part should be in RAX, and its high part in RDX.
```

Comment:
Minor: You might want to frame the above comment on lines similar to the comment on line 2211 since LSRA doesn't guarantee that always.

eg.
// To avoid reg move would like to have op1's low part in RAX and high part in RDX.